### PR TITLE
add "profiler" and "optimized-compiler-builtins" option coverage for ci-rustc

### LINKED ISSUE #133675: download-rustc fails to detect when profiler settings differ from local settings
See https://github.com/rust-lang/rust/pull/133499#issuecomment-2508901283 for context: the  x86_64-gnu-aux runner builds rustc without profiling support. The CI build that is downloaded via download-rustc *does* have profiling support. This means if a test relies on profiling support, it will incorrectly pass on  x86_64-gnu-aux if download-rustc is used, only to fail later on an unrelated PR when download-rustc cannot be used.

bootstrap tries to ensure that all the settings are the same between the local configuration and the prebuilt rustc, but that does not seem to fully work.

Cc @onur-ozkan

### PR DESCRIPTION
**Pull request by onur-ozkan**
Adds "profiler" and "optimized-compiler-builtins" option coverage in CI-rustc config compatibility check. 

Resolves #133675

### CODE REVIEW

File: src/bootstrap/src/core/config/config.rs [hunk 1]
```diff
@@ -3123,6 +3125,26 @@ fn check_incompatible_options_for_ci_rustc(
         };
     }
 
+    err!(
+        current_config_toml.build.as_ref().and_then(|b| b.profiler),
+        ci_config_toml.build.as_ref().and_then(|b| b.profiler),
+        "build"
+    );
```
[the remark this answers is not in the document]
Comment (jieyouxu) [reply src/bootstrap/src/core/config/config.rs:3132]: I think that should probably also be checked.

**Review by jieyouxu — APPROVED** [review]
Thanks, you can r=me after also adding a check for optimized compiler builtins.

File: src/bootstrap/src/core/config/config.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (onur-ozkan) [reply src/bootstrap/src/core/config/config.rs]: Makes sense.

**Review by jieyouxu — APPROVED** [review]

**Pull request merged**